core/src/main/java/hudson/Functions.java

```diff
@@ -233,6 +233,10 @@ public static void initPageVariables(JellyContext context) {
 
         // The path starts with a "/" character but does not end with a "/" character.
         context.setVariable("rootURL", rootURL);
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.setCachedRootUrl(jenkins.getRootUrlFromRequest());
```

Comment:
The new version has the Restricted annotation, fixed.
A security impact of initPageVariables() needs to be elaborated BTW.